packages/flutter_tools/test/src/common.dart

```diff
@@ -184,6 +184,9 @@ void test(String description, FutureOr<void> Function() body, {
     // We don't support "timeout"; see ../../dart_test.yaml which
     // configures all tests to have a 15 minute timeout which should
     // definitely be enough.
+    // Specifying a timeout here as well enables running test files
+    // standalone without `flutter test`.
+    timeout: const Timeout(Duration(minutes: 15)),
```

Comment:
DBC

It would be confusing to change the number in `dart_test.yaml`, and for that change to have no effect.

Reply:
We can leave cross-comments here and there.

If I remove this line running a test with `dart path/to/my_test.dart` times out. Running `flutter test` instead is slower due to having to precompile some package:test machinery.

I'm happy to revert this before I land this PR if you'd really don't want to support the `dart path/to/my_test.dart`.

Reply:
> If I remove this line running a test with `dart path/to/my_test.dart` times out.

What's the root cause of that? Is the default timeout too short?

> support the `dart path/to/my_test.dart`.

If it isn't tested on CI, it will likely break some other way even if this line is added. If this is something that you'd like to keep working, you'd need to add a test to CI for it. Given the size of this change, that might be better to do as a separate PR.

Reply:
The root cause is individual tests having a timeout duration of 30 seconds by default.

```
$ dart /Users/dacoharkes/flt/engine/flutter/packages/flutter_tools/test/integration.shard/native_assets_test.dart
00:00 +0: (setUpAll)

00:00 +0: flutter run hot reload and hot restart with native assets flutter-tester debug

00:16 +1: flutter run hot reload and hot restart with native assets macos debug

00:41 +2: flutter run with native assets macos profile

01:11 +2 -1: flutter run with native assets macos profile [E]

  TimeoutException after 0:00:30.000000: Test timed out after 30 seconds. See https://pub.dev/packages/test#timeouts

  package:test_api/src/backend/invoker.dart 338:28  Invoker._handleError.<fn>
  dart:async/zone.dart 1391:47                      _rootRun
  dart:async/zone.dart 1301:19                      _CustomZone.run
  package:test_api/src/backend/invoker.dart 336:10  Invoker._handleError
  package:test_api/src/backend/invoker.dart 291:9   Invoker.heartbeat.<fn>.<fn>
  dart:async/zone.dart 1399:13                      _rootRun
  dart:async/zone.dart 1301:19                      _CustomZone.run
  package:test_api/src/backend/invoker.dart 290:38  Invoker.heartbeat.<fn>
  dart:async-patch/timer_patch.dart 18:15           Timer._createTimer.<fn>
  dart:isolate-patch/timer_impl.dart 398:19         _Timer._runTimers
  dart:isolate-patch/timer_impl.dart 429:5          _Timer._handleMessage
  dart:isolate-patch/isolate_patch.dart 184:12      _RawReceivePort._handleMessage
  

01:11 +2 -1: flutter run with native assets macos release

[...]

$ flutter test /Users/dacoharkes/flt/engine/flutter/packages/flutter_tools/test/integration.shard/native_assets_test.dart
[ ... works fine ]
```

> > support the `dart path/to/my_test.dart`.
> 
> If it isn't tested on CI, it will likely break some other way even if this line is added. If this is something that you'd like to keep working, you'd need to add a test to CI for it. Given the size of this change, that might be better to do as a separate PR.

That might be a bit overkill, testing that we can run a test as a standalone dart file instead of with `flutter test`. I'll revert the change.